### PR TITLE
feat(modal-checkout): allow anonymous purchase for registered email

### PR DESCRIPTION
**Pull request by miguelpeixe**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-theme/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Moves a logic that is [currently placed behind RAS](https://github.com/Automattic/newspack-plugin/blob/fcad059abbe0314ba8a6f3eaaf92d5c96350e248/includes/reader-activation/class-reader-activation.php#L89-L90) but is designed for the modal checkout.

If an anonymous purchase is made for a registered email, it should be attached to that account regardless of them being authenticated.

### How to test the changes in this Pull Request:

1. Checkout this branch and https://github.com/Automattic/newspack-plugin/pull/2781
2. Make sure RAS is toggled off (`$ wp option delete newspack_reader_activation_enabled`)
3. Disable the "Allow customers to create an account during checkout" option from the "Accounts & Privacy" section of WC:
image

4. Go through the regular checkout flow anonymously and use an existing email
5. Confirm the order goes through and in the dashboard the order is not assigned to any user (shown as "guest" customer)
6. Also anonymously, go through the modal checkout flow (either through the Checkout Button block or Donate block) using an existing email
7. Confirm the order has been placed and assigned to the user
8. Visit `/my-account` and confirm you're still logged out

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Review by adekbadek — COMMENTED** [review]
> Confirm you're prompted to log in, as it's available through this checkout UI

Can't get this part to work, tried with different configurations of the guest checkout options. Tested by buying a subscription in a an anonymous session, using an existing user's email. What is the correct configuration on WC guest checkout for this to work? And at which point should the login prompt appear?

image

File: includes/class-modal-checkout.php [hunk 1]
```diff
@@ -973,6 +977,50 @@ public static function order_button_text( $text ) {
 		return $text;
 	}
 
+	/**
+	 * If a reader tries to make a recurring donation with an email address that
```
Comment (leogermani) [inline includes/class-modal-checkout.php:981]:
```suggestion
	 * If a reader tries to make a purchase with an email address that
```

**Review by leogermani — COMMENTED** [review]
It works, I left one minor fix in the docs

But I'm still struggling to understand why this should be a modal checkout thing and not a RAS.

On one hand, I understand that the modal checkout should be Woocommerce friendly and compatible with its features

On the other hand, I thought anything that would automatically register a user in WP should only happen under RAS (or if explicitly enabled in woocommerce in the cases it does it)

**Comment by miguelpeixe** [comment]
> But I'm still struggling to understand why this should be a modal checkout thing and not a RAS.

The modal checkout doesn't currently support creating an account mid-checkout and we have publishers outside of RAS using it. This creates a scenario in which existing, but logged-out, customers get stuck while making a purchase.

Context: p1701216656275889-slack-C0390J7K0KY

>  I thought anything that would automatically register a user in WP should only happen under RAS

That is still the case. These hooks are not to create an account. It attaches the order to the existing email if it exists.

**Review by leogermani — APPROVED** [review]
thanks for the context

**Pull request merged**